System/Process/Internals.hs

```diff
@@ -414,6 +429,8 @@ foreign import ccall unsafe "runInteractiveProcess"
         -> Ptr FD
         -> Ptr FD
         -> Ptr FD
+        -> Ptr CGid
```

Comment:
Given that a CGid or CUid is just an `int`, I think this can be done without the pointer indirection, by using `-1` to represent "not provided."